File: js_solutions/JZ45.js
function IsContinuous(numbers)
{
    // write code here
    let max = 0, min = 14;
    if(numbers.length != 5) 
        return false;
    for(let i = 0; i < numbers.length; i++) {
        // 数字越界，false
        if(numbers[i] < 0 || numbers[i] > 13)
            return false;
        if(numbers[i] === 0) {
            continue;
        }
        // 数组中出现重复的数字，false
        const lastIndex = numbers.lastIndexOf(numbers[i]);
        if(lastIndex != -1 && lastIndex != i)
            return false;
        // 确定最大值和最小值的范围
        max = Math.max(max, numbers[i]);
        min = Math.min(min, numbers[i]);
    }
    return max - min < 5 ;
}

File: java_solutions/JZ45.java
import java.util.*;
public class Solution {
    public boolean isContinuous(int [] numbers) {
        HashSet<Integer> set = new HashSet<>();
        int min = 14, max = 0;
        if(numbers.length != 5) {
            return false;
        }
        for(int number :  numbers) {
            if(number < 0 || number > 13) {
                return false;
            }
            // 大王小王
            if(number == 0) {
                continue;
            }
            if(set.contains(number)){
                return false;
            }
            set.add(number);
            min = Math.min(min, number);
            max = Math.max(max, number);
            set.add(number);
        }
        return max - min < 5;
    }
}